<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="m4assembly">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.00014012 -1.2866E-05 -0.067235"
        rpy="0 0 0" />
      <mass
        value="1.054" />
      <inertia
        ixx="0.00066925"
        ixy="-1.7138E-07"
        ixz="8.4179E-07"
        iyy="0.0060916"
        iyz="-5.4732E-08"
        izz="0.006372" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.75294 0.75294 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="rear_left_hip">
    <inertial>
      <origin
        xyz="0.032048263261567 0.00148857273521091 0.000801285389170403"
        rpy="0 0 0" />
      <mass
        value="0.0631744337027232" />
      <inertia
        ixx="1.00187470871694E-05"
        ixy="1.26346518144828E-06"
        ixz="-1.71088892144535E-13"
        iyy="1.47423630145446E-05"
        iyz="-3.33426314544602E-14"
        izz="2.18314090226691E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_left_hip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_left_hip.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_left_hip_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="-0.2231 0.0323 -0.02605"
      rpy="1.5708 -0.0099006 1.5708" />
    <parent
      link="base_link" />
    <child
      link="rear_left_hip" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.571"
      effort="10.5"
      velocity="1.57" />
  </joint>
  <link
    name="rear_left_leg">
    <inertial>
      <origin
        xyz="-3.41455978458693E-06 0.0865094820933296 -0.00425258484840052"
        rpy="0 0 0" />
      <mass
        value="0.208886814057311" />
      <inertia
        ixx="7.70084638154863E-05"
        ixy="9.13752456418355E-09"
        ixz="3.41558069671286E-07"
        iyy="5.22430908180149E-05"
        iyz="-3.49227338347394E-08"
        izz="5.21652155907927E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_left_leg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_left_leg.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_left_leg_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.0611 -0.006895 -5E-05"
      rpy="1.5708 -1.5526 1.5708" />
    <parent
      link="rear_left_hip" />
    <child
      link="rear_left_leg" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <link
    name="rear_left_wheel">
    <inertial>
      <origin
        xyz="-1.2835011736767E-06 2.34490590667047E-07 -0.0298825868697071"
        rpy="0 0 0" />
      <mass
        value="0.447367748033206" />
      <inertia
        ixx="0.00104268899323013"
        ixy="-3.91603340525233E-08"
        ixz="3.60415118706037E-11"
        iyy="0.00104263881447407"
        iyz="-3.26123864995785E-11"
        izz="0.00200352361648073" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_left_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_left_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_left_wheel_joint"
    type="continuous">
    <origin
      xyz="0 0.162 0.011129"
      rpy="-3.1416 0 0.018227" />
    <parent
      link="rear_left_leg" />
    <child
      link="rear_left_wheel" />
    <axis
      xyz="0 0 -1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="rear_left_blade">
    <inertial>
      <origin
        xyz="-3.30314530727627E-08 -8.23060748793014E-05 0.0082946342231609"
        rpy="0 0 0" />
      <mass
        value="0.00610274937825579" />
      <inertia
        ixx="4.75952063199721E-06"
        ixy="-3.96883337183772E-08"
        ixz="-3.93240380060884E-10"
        iyy="1.57361545543065E-07"
        iyz="-4.65846533883739E-08"
        izz="4.86189149512478E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_left_blade.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_left_blade.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_left_blade_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0 0.162 0.061129"
      rpy="-0.0099006 0 0.018227" />
    <parent
      link="rear_left_leg" />
    <child
      link="rear_left_blade" />
    <axis
      xyz="0 -0.0099004 0.99995" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="rear_right_hip">
    <inertial>
      <origin
        xyz="0.0320482634870919 0.00148857274187802 -0.000820989548367512"
        rpy="0 0 0" />
      <mass
        value="0.0631744341018766" />
      <inertia
        ixx="1.00187470810585E-05"
        ixy="1.26346518895748E-06"
        ixz="-1.58209935155058E-13"
        iyy="1.47423629783338E-05"
        iyz="-2.99342928986296E-14"
        izz="2.18314089935792E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_right_hip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_right_hip.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_right_hip_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="-0.2235 -0.0323 -0.02605"
      rpy="1.5708 -0.0032479 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="rear_right_hip" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.571"
      effort="10.5"
      velocity="1.57" />
  </joint>
  <link
    name="rear_right_leg">
    <inertial>
      <origin
        xyz="3.41455156416259E-06 0.0865094821571721 0.00425258483326613"
        rpy="0 0 0" />
      <mass
        value="0.208886814304404" />
      <inertia
        ixx="7.70084638389827E-05"
        ixy="-9.13752631729426E-09"
        ixz="3.41558058672973E-07"
        iyy="5.22430908411789E-05"
        iyz="3.49227166953638E-08"
        izz="5.21652156031499E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_right_leg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_right_leg.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_right_leg_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.0611 -0.006895 0"
      rpy="3.1416 1.5708 0" />
    <parent
      link="rear_right_hip" />
    <child
      link="rear_right_leg" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="rear_right_wheel">
    <inertial>
      <origin
        xyz="-1.78659846361562E-08 -2.01874887645337E-06 0.0255799203790625"
        rpy="0 0 0" />
      <mass
        value="0.447367750743676" />
      <inertia
        ixx="0.0010426206896193"
        ixy="-2.73561891555305E-10"
        ixz="3.48030847023143E-11"
        iyy="0.00104270712224184"
        iyz="3.22928670574781E-11"
        izz="0.00200352362095162" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_right_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_right_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_right_wheel_joint"
    type="continuous">
    <origin
      xyz="0 0.162 -0.015431"
      rpy="3.1416 0 0" />
    <parent
      link="rear_right_leg" />
    <child
      link="rear_right_wheel" />
    <axis
      xyz="0 0 -1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="rear_right_blade">
    <inertial>
      <origin
        xyz="2.93887484903443E-08 6.79423276528368E-08 0.00399237045178649"
        rpy="0 0 0" />
      <mass
        value="0.00610270077025495" />
      <inertia
        ixx="4.75712306507452E-06"
        ixy="-1.12007295746605E-07"
        ixz="2.24623999105101E-13"
        iyy="1.59284414569248E-07"
        iyz="3.54828245352755E-13"
        izz="4.86233952224275E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/rear_right_blade.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/rear_right_blade.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="rear_right_blade_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0 0.162 -0.065431"
      rpy="3.1416 0 0" />
    <parent
      link="rear_right_leg" />
    <child
      link="rear_right_blade" />
    <axis
      xyz="0 0 1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="front_left_hip">
    <inertial>
      <origin
        xyz="0.0320482633471696 0.00148857273674285 0.00169341200883941"
        rpy="0 0 0" />
      <mass
        value="0.0631744338743183" />
      <inertia
        ixx="1.00187470896772E-05"
        ixy="1.26346518496343E-06"
        ixz="-1.75127210359815E-13"
        iyy="1.474236301921E-05"
        iyz="-3.22758580937248E-14"
        izz="2.18314090243211E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_left_hip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_left_hip.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_left_hip_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.22099 0.0323 -0.02605"
      rpy="1.5708 -0.006693 1.5708" />
    <parent
      link="base_link" />
    <child
      link="front_left_hip" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.571"
      effort="10.5"
      velocity="1.57" />
  </joint>
  <link
    name="front_left_leg">
    <inertial>
      <origin
        xyz="-3.41455576607095E-06 0.0865094821265606 -0.00425258483916741"
        rpy="0 0 0" />
      <mass
        value="0.208886814202469" />
      <inertia
        ixx="7.70084638282559E-05"
        ixy="9.13752541336675E-09"
        ixz="3.41558064254658E-07"
        iyy="5.22430908306995E-05"
        iyz="-3.49227240782923E-08"
        izz="5.21652155975951E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_left_leg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_left_leg.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_left_leg_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.0611 -0.006895 0.0025144"
      rpy="-1.5708 -1.5708 -1.5708" />
    <parent
      link="front_left_hip" />
    <child
      link="front_left_leg" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <link
    name="front_left_wheel">
    <inertial>
      <origin
        xyz="1.77235233511999E-08 -2.01879895370127E-06 -0.0284894545323338"
        rpy="0 0 0" />
      <mass
        value="0.447367750292981" />
      <inertia
        ixx="0.00104262069065618"
        ixy="2.74220212176939E-10"
        ixz="3.54050122778484E-11"
        iyy="0.00104270712119543"
        iyz="-3.25695335683767E-11"
        izz="0.00200352362063109" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_left_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_left_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_left_wheel_joint"
    type="continuous">
    <origin
      xyz="0 0.162 0.012522"
      rpy="3.1416 0 0" />
    <parent
      link="front_left_leg" />
    <child
      link="front_left_wheel" />
    <axis
      xyz="0 0 -1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="front_left_blade">
    <inertial>
      <origin
        xyz="-1.18411884182483E-08 -1.5997300897852E-07 0.0069019562259629"
        rpy="0 0 0" />
      <mass
        value="0.0061028301426362" />
      <inertia
        ixx="4.65427221541224E-06"
        ixy="6.89623690398127E-07"
        ixz="-5.02014385573018E-13"
        iyy="2.62287309333734E-07"
        iyz="-7.87853177369262E-13"
        izz="4.86249379657286E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_left_blade.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_left_blade.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_left_blade_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0 0.162 0.062522"
      rpy="0 0 0" />
    <parent
      link="front_left_leg" />
    <child
      link="front_left_blade" />
    <axis
      xyz="0 0 1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="front_right_hip">
    <inertial>
      <origin
        xyz="0.0320482634665248 0.0014885727412563 -0.00131311618414334"
        rpy="0 0 0" />
      <mass
        value="0.0631744340641276" />
      <inertia
        ixx="1.00187470815544E-05"
        ixy="1.26346518823477E-06"
        ixz="-1.57692513882067E-13"
        iyy="1.47423629788781E-05"
        iyz="-2.96148209881054E-14"
        izz="2.18314089943886E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_right_hip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_right_hip.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_right_hip_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.22099 -0.0323 -0.02605"
      rpy="1.5708 -0.0046828 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="front_right_hip" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="1.571"
      effort="10.5"
      velocity="1.57" />
  </joint>
  <link
    name="front_right_leg">
    <inertial>
      <origin
        xyz="3.41455365766163E-06 0.0865094821436897 0.00425258483703411"
        rpy="0 0 0" />
      <mass
        value="0.208886814244098" />
      <inertia
        ixx="7.70084638339796E-05"
        ixy="-9.13752605384622E-09"
        ixz="3.4155806167583E-07"
        iyy="5.2243090836002E-05"
        iyz="3.49227209398553E-08"
        izz="5.21652156003534E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_right_leg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_right_leg.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_right_leg_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0.0611 -0.006895 -0.0021644"
      rpy="1.5708 1.5656 -1.5708" />
    <parent
      link="front_right_hip" />
    <child
      link="front_right_leg" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-0.5"
      upper="0.5"
      effort="10.5"
      velocity="1.57" />
  </joint>
  <link
    name="front_right_wheel">
    <inertial>
      <origin
        xyz="1.28310223720821E-06 2.3448596928044E-07 0.0255837574129385"
        rpy="0 0 0" />
      <mass
        value="0.447367750640603" />
      <inertia
        ixx="0.00104268899474349"
        ixy="3.91619025865096E-08"
        ixz="3.50659148166867E-11"
        iyy="0.00104263881744075"
        iyz="3.24293757397963E-11"
        izz="0.00200352362110246" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_right_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_right_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_right_wheel_joint"
    type="continuous">
    <origin
      xyz="0 0.162 -0.015427"
      rpy="-3.1416 0 -0.0052077" />
    <parent
      link="front_right_leg" />
    <child
      link="front_right_wheel" />
    <axis
      xyz="0 0 -1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>
  <link
    name="front_right_blade">
    <inertial>
      <origin
        xyz="6.96858681763831E-08 7.67517420813268E-07 0.0039962219905107"
        rpy="0 0 0" />
      <mass
        value="0.00610290216434297" />
      <inertia
        ixx="4.75885051711651E-06"
        ixy="7.08408517084622E-08"
        ixz="-8.40829933326352E-14"
        iyy="1.57650099061995E-07"
        iyz="8.23720329397051E-13"
        izz="4.86243263794242E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/front_right_blade.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://simulation_world/meshes/col/front_right_blade.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="front_right_blade_joint"
    type="fixed" dont_collapse="true">
    <origin
      xyz="0 0.162 -0.065427"
      rpy="-3.1416 0 -0.0052077" />
    <parent
      link="front_right_leg" />
    <child
      link="front_right_blade" />
    <axis
      xyz="0 0 1" />
    <limit
      effort="200"
      velocity="200" />
  </joint>

    <link name="front_realsense" />Z

  <link name="front_realsense_lens">
    <visual>
      <origin xyz="0.02 0 0" rpy="1.5707963267948966 0 1.5707963267948966" />
      <geometry>
        <mesh filename="package://realsense2_description/meshes/d435.dae" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.75294 0.75294 0.75294 1" />
      </material>
    </visual>
  </link>

  <joint type="fixed" name="front_realsense_lens_joint">
    <!-- Offset the camera 2cm backwards and 1cm up -->
    <origin xyz="0.24099 0 0" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="front_realsense_lens" />
  </joint>
  <joint type="fixed" name="front_realsense_joint">
    <origin xyz="0.025 0 0" rpy="0 0 0" />
    <parent link="front_realsense_lens" />
    <child link="front_realsense" />
  </joint>

</robot>
